AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -132,9 +132,17 @@ public class LaunchRulesEngine {
                     eventData = modifiedEventData
 
                 case LaunchRulesEngine.CONSEQUENCE_TYPE_DISPATCH:
-                    if let unwrappedDispatchCount = dispatchChainCount, unwrappedDispatchCount >= LaunchRulesEngine.MAX_CHAINED_CONSEQUENCE_COUNT {
-                        Log.trace(label: LOG_TAG, "(\(self.name)) : Unable to process a DispatchConsequence Event, max dispatch consequences met for this event, \(event)")
-                        continue
+
+                    if let unwrappedDispatchCount = dispatchChainCount {
+                        // Keep track of triggering event in edge case of event being dispatched multiple times
+                        // This only applies to events created from this dispatch consequence, hence the check
+                        // for a null 'dispatchChainCount' prior to adding back to dictionary
+                        dispatchChainedEventsCount[event.id] = LaunchRulesEngine.MAX_CHAINED_CONSEQUENCE_COUNT
```

Comment:
Do we need this logic? This means irrespective of the `LaunchRulesEngine.MAX_CHAINED_CONSEQUENCE_COUNT` the chain length can not be more than 2.

Reply:
This means that a "link" in the chain will not get processed more than once. The chain length will still be MAX_CHAINED_CONSEQUENCE_COUNT. This is catching the case where an event dispatched by this consequence will only trigger a dispatch consequence once. If the same event is reprocessed, then it will not dispatch more events. As I'm checking if `dispatchChainCount` is not null, only events which were originally created by the dispatch consequence are handled, but not events which originate else where.

Reply:
This becomes little more complicated than the use-case we started this thread of changes with. Can we instead keep the count for the original chain link for this triggering event? 
```
e.g. MAX_CHAINED_CONSEQUENCE_COUNT = 2
Let's assume E1 triggered two dispatch consequences E2 and E3, E2 triggered E4 
dispatchChainedEventsCount should look like:
uuid E2: 1
uuid E3: 1
uuid E4: 2
E4 cannot trigger new dispatch consequences as its chain maxed out, 
but a replay of E2 and E3 is allowed and they will generate at most one additional event each.
```

Reply:
I feel allowing a tree structure like that is more confusing. It's not clear why E2 can be replayed while E4 cannot. 

For example, if MAX_CHAINED_CONSEQUENCE_COUNT = 5, and there are rules which chain dispatch consequences:
E0 -> E1 (1) -> E2 (2) -> E3 (3) -> E4 (4) -> E5 (5)
A replay of E1 produces:
E1 (2) -> E2a (3) -> E3a (4) -> E4a (5)
Another replay of E1 produces:
E1 (3) -> E2b (4) -> E3b (5)
Etc.

This behavior is not intuitive as users have little in site into the internal count we assign to each event. What I'm advocating for is preventing the replay of events created from the dispatch consequence.

Reply:
I feel we should use this count to only limits events automatically generated by rules in response to a single `MobileCore.dispatch` API call. Let us treat the replayed events as new events to make the behaviour predictable. 
We should take a second pass when we look into #688

Reply:
@praveek agreed, will simplify this logic in Android too.